### PR TITLE
Resolves toin0u/DigitalOceanV2#206

### PR DESCRIPTION
**Pull request by tspencer244**
These are the changes to make the library PHP 7.1+. Don't see how HHVM support can continue. Recommend tagging this as release 3.0.0.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Actually resolves https://github.com/toin0u/DigitalOceanV2/issues/195 I think?

**Comment by GrahamCampbell** [comment]
Could you send the new GuzzleHttpAdapterSpec as a separate PR please? That can get merged right away.

File: composer.json [hunk 1]
```diff
@@ -27,22 +27,20 @@
...
-        "kriswallsmith/buzz"        : "~0.10",
-        "guzzle/guzzle"             : "~3.7",
-        "guzzlehttp/guzzle"         : "~5.0|~6.0",
-        "phpspec/phpspec"           : "~2.4",
-        "friendsofphp/php-cs-fixer" : "~1.11",
-        "phpdocumentor/phpdocumentor": "2.*"
+        "kriswallsmith/buzz"        : "~0.17",
+        "guzzlehttp/guzzle"         : "~6.0",
+        "phpspec/phpspec"           : "~5.0",
+        "friendsofphp/php-cs-fixer" : "~2.13"
```
Comment (GrahamCampbell) [inline composer.json]: This won't actually work. They have different config formatting. I'd recommend we actually drop php-cs-fixer from require-dev, and just use https://styleci.io/.

Comment (GrahamCampbell) [reply composer.json]: I've ripped out php-cs-fixer from master.

Comment (tspencer244) [reply composer.json]: Have merged upstream so this should be resolved now

File: src/Api/Droplet.php [hunk 2]
```diff
@@ -163,7 +163,7 @@ public function create($names, $region, $size, $image, $backups = false, $ipv6 =
         $droplet = json_decode($droplet);
 
         if (is_array($names)) {
-            return array_map(function ($droplet) {
+            return array_map(function ($droplet) use ($wait, $waitTimeout) {
```
Comment (tspencer244) [inline src/Api/Droplet.php]: Seems like creating more than one droplet would have ignored the wait parameters?

File: spec/DigitalOceanV2/Api/DropletSpec.php [hunk 3]
```diff
@@ -289,18 +348,49 @@ public function it_returns_the_created_droplet_entity_with_ssh_keys($adapter)
             ->shouldReturnAnInstanceOf('DigitalOceanV2\Entity\Droplet');
     }
 
-    public function can_create_multiple_droplets_at_the_same_time($adapter)
+    /**
+     * @param \DigitalOceanV2\Adapter\AdapterInterface $adapter
+     */
+    public function it_can_create_multiple_droplets_at_the_same_time($adapter)
```
Comment (tspencer244) [inline spec/DigitalOceanV2/Api/DropletSpec.php]: Not sure if this test was intentionally left in a state where it wouldn't run but I tried to fix it

Comment (GrahamCampbell) [reply spec/DigitalOceanV2/Api/DropletSpec.php]: Please try and keep this PR atomic please, so we can merge fixes we need, without coupling everything to this PR. :)

**Comment by tspencer244** [comment]
> Could you send the new GuzzleHttpAdapterSpec as a separate PR please? That can get merged right away.

See #219 for the separate PR

File: spec/DigitalOceanV2/Api/DropletSpec.php (hunk 3 above)
Comment (tspencer244) [reply spec/DigitalOceanV2/Api/DropletSpec.php]: Sorry about that, will try to keep PR more atomic from now on

**Comment by GrahamCampbell** [comment]
I can't merge this in it's current state, because it breaks support for the old version of buzz, and renames classes, breaking existing code.

**Comment by GrahamCampbell** [comment]
Hmmm, will make master 3.x, and fix the version of buzz needed in the docs for 2.x.

Thanks for the PR. 🍻 🍻 🍻

**Pull request merged**